Sources/COWMacros/SyntaxExtensions.swift

```diff
@@ -129,6 +137,107 @@ extension FunctionDeclSyntax {
         }
       }
     }
+    return false
+  }
+  
+  var isMutating: Bool {
+    modifiers.contains(where: { $0.name.tokenKind == .keyword(.`mutating`) })
+  }
+  
+  func returnTypeEquals(to type: String) -> Bool {
+    guard let returnClause = signature.returnClause,
+          let returnType = returnClause.type.as(IdentifierTypeSyntax.self) else {
+      return type == "Void"
+    }
+    return returnType.name.trimmed.text == type
+  }
+  
+  // Equatable:
+  // static func == (lhs: Self, rhs: Self) -> Bool
+  func likelyToConformToEquatable(for structDecl: StructDeclSyntax) -> Bool {
+    guard isStatic else {
+      return false
+    }
+    guard name.tokenKind == .binaryOperator("=="),
+          returnTypeEquals(to: "Bool") else {
+      return false
+    }
+    let params = signature.parameterClause.parameters
+    guard params.count == 2 else {
+      return false
+    }
+    guard let lhs = params.first, let rhs = params.last else {
+      return false
+    }
+    guard lhs.firstName.trimmed.text == "lhs",
+          rhs.firstName.trimmed.text == "rhs",
+          let lhsType = lhs.type.as(IdentifierTypeSyntax.self),
+          let rhsType = rhs.type.as(IdentifierTypeSyntax.self),
+          lhsType.name.tokenKind == rhsType.name.tokenKind else {
+      return false
+    }
+    // Accept either `Self` or the name of the struct.
+    guard lhsType.name.tokenKind == .keyword(.`Self`) ||
+            lhsType.name.tokenKind == structDecl.name.tokenKind else {
+      return false
+    }
+    return true
+  }
+  
+  // Hashable:
+  // func hash(into hasher: inout Hasher)
+  var likelyToConformToHashable: Bool {
```

Comment:
Explicit `internal` access control is recommended. This convention is learned from the Swift Standard Library.